### PR TITLE
Feature/add test cd

### PR DESCRIPTION
**Pull request by idelcano**
### :pushpin: References

-   **Issue:** Coding dojo

### :memo: Implementation
Added some test for user-permission script (In this case we change a bit the workflow and I did a manual test also to be sure that nothing is broken).

    Should ignore user if the user is in the datastore excluded users list
    Should ignore user if the user has valid roles
    Should fix user if the user has invalid authorities
    Should throw exception if the user don't have a valid user template group and fix is disabled
    Should fix add minimal group if a user dont have any control template user group
    Should push fixed usergroup if the user has no template usergroups
    Should apply minimal control template group if a user dont have any control template user group

We could continue adding test due has a lot of corner case to test.

### :fire: Notes to the tester

yarn start usermonitoring run-permissions-fixer --config-file config.json

### CODE REVIEW

File: src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    permissionFixerInvalidTemplateGroupsExtended.group = { id: "notValidGroup", name: "not valid group" };
+    clonedInvalidTemplateAuthorities = copyObject(permissionFixerInvalidTemplateGroupsExtended);
+});
+
+function getMetadataConfig(): PermissionFixerMetadataConfig {
+    return baseMetadataConfig;
+}
+
+function getMetataConfiThrowInvalidUsergroupException(): PermissionFixerMetadataConfig {
+    return configThrowInvalidUsergroupException;
```
Comment (xurxodev) [inline src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts:163]: This function is not necessary and generate noise, you can use the variable direcly

File: src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+}
+
+function getMetataConfigWithExcludedUser(): PermissionFixerMetadataConfig {
+    return configWithUserExcluded;
+}
+
+function getWrongMinimalUserGroupConfig(): PermissionFixerMetadataConfig {
+    return configWithWrongMinimalGroup;
```
Comment (xurxodev) [inline src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts:171]: This function is not necessary and generate noise, you can use the variable direcly

File: src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    configThrowInvalidUsergroupException = copyObject(baseMetadataConfig);
+    clonedTemplateAuthorities = copyObject(permissionFixerTemplateGroupsExtended);
+    const permissionFixerInvalidTemplateGroupsExtended: PermissionFixerTemplateGroupExtended = copyObject(
+        permissionFixerTemplateGroupsExtended[0]
+    );
+    permissionFixerInvalidTemplateGroupsExtended.group = { id: "notValidGroup", name: "not valid group" };
+    clonedInvalidTemplateAuthorities = copyObject(permissionFixerInvalidTemplateGroupsExtended);
+});
+
+function getMetadataConfig(): PermissionFixerMetadataConfig {
```
[the remark this answers is not in the document]
Comment (idelcano) [reply src/domain/usecases/user-monitoring/permission-fixer/__tests__/RunUserPermissionUseCase.specs.ts]: All fixed

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @idelcano 

It seems that the last commit broke the github action

Can you review it?

**Comment by idelcano** [comment]
Hi @xurxodev, thanks for the review.

I saw it but the error originates from development:

Error: Failed to load url @dhis2/d2-i18n (resolved id: @dhis2/d2-i18n) in /home/runner/work/d2-tools/d2-tools/src/capture-core/metaData/helpers/EventProgram/getStageForEventProgram.js. Does the file exist?

All last PR was failing:
https://github.com/EyeSeeTea/d2-tools/pulls

I can check it, but I think this is a different issue, not related to my branch. Maybe it makes more sense to wait and merge a fixed development branch.

**Comment by xurxodev** [comment]
@idelcano I understand that it's possible that the origin of the bug is not in your branch, however you merged with development bringing the changes from development.

Probably it was an error of someone to merge the old PR in development with this error, but I can't approve this PR if the GitHub action fails. 

Please fix the bug, probably talking with who introduced the bug. Let me know when it's fixed and I can to review the PR.

**Review by xurxodev — APPROVED** [review]
thanks @idelcano
